lib/index.js

```diff
@@ -192,6 +192,42 @@ module.exports = function(session) {
 		});
 	};
 
+	MySQLStore.prototype.touch = function(session_id, data, cb) {
+
+		debug_log('Touching session: ' + session_id);
+
+		var expires;
+
+		if (data.cookie && data.cookie.expires) {
+			expires = data.cookie.expires;
+		} else {
+			expires = new Date(Date.now() + this.options.expiration);
+		}
```

Comment:
I think the touch method should always set a new value for expires. If the old value is used, the session will expire eventually even if touch is used.